File: ObjC/2_View전환/SampleApp/ViewController.m
#import "ViewController.h"


// 화면 전환 하는 방법
//  - Android
//    => Activity 전환
//       Fragment 전환

//  - iOS
//    => 화면을 전환하는 다양한 방법

// UIWindow
//    - rootViewController: ViewController
//                              - UIView* view
//     1) Interface Builder

//     2) Storyboard

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view.
}

// view에 새로운 view 추가하는 함수
//  - addSubView
//  - removeFromSuperview

- (IBAction)onTouch:(id)sender {
  
  [UIView beginAnimations:nil context:nil];
  
  [UIView setAnimationDuration:1];
  [UIView setAnimationTransition:UIViewAnimationTransitionCurlUp
                         forView:self.view
                           cache:YES];
  
  [self.view addSubview:_childView];
  
  
  [UIView commitAnimations];
}

- (IBAction)onTouch2:(id)sender {
  [_childView removeFromSuperview];
}





@end

File: Swift/6_NavigationController/SampleApp/ViewController.swift

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view.
  }


  @IBAction func onTouch(_ sender: Any) {
    // UINavigation Controller를 코드에서 사용하는 방법
    let naviController = UINavigationController(rootViewController:
                                                  FirstController.create())
    
    naviController.modalPresentationStyle = .fullScreen
    present(naviController, animated: true)
  }
}

